common-lib/src/cmr/common/hash_cache.clj

```diff
@@ -0,0 +1,57 @@
+(ns cmr.common.hash-cache
+  "Defines the core hash caching protocol for the CMR."
+  (:require
+   [cmr.common.cache :as cache]))
+
+(defn context->cache
+  "Get the cache for the given key from the context"
+  [context cache-key]
+  (cache/context->cache context cache-key))
+
+(defprotocol CmrHashCache
+  "Defines a protocol used for caching data using hashes."
+  (get-map
```

Comment:
get-map and get-keys have the same variables, how are they going to differ? Would be good to have a comment for get-map

Reply:
Get map gets the entire hash map back, whereas get-keys only gets you the field names (hash keys) back.

Reply:
I will add a comment.